### PR TITLE
move internal/net/tcp package to internal/net package and support unix domain socket

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <kpango@vdaas.org>

### Description:
This PR includes 6 changes
1. merge internal/net/tcp package to internal/net
2. create new package internal/net/control for socket configuration
3. unix domain socket support
4. change helm chart related to 1&2&3
5. update Go version 1.16 for checking tcp connectivity for future go update
6. patch update k8s dependencies

### How Has This Been Tested?:
I executed make e2e command and checked all existing test has been passed.

### Environment:

- Go Version: 1.15.8
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kpango** [comment]
@vdaas/core @vdaas/sre @vdaas/set This pr is now ready for review, please review this PR

File: internal/net/net_test.go [hunk 1]
```diff
@@ -277,276 +216,276 @@ func TestDial(t *testing.T) {
...
+// 				test.checkFunc = defaultCheckFunc
+// 			}
+//
+// 			got := IsIPv4(test.args.addr)
+// 			if err := test.checkFunc(test.want, got); err != nil {
+// 				tt.Errorf("error = %v", err)
+// 			}
+// 		})
+// 	}
+// }
```
Comment (vankichi) [inline internal/net/net_test.go:488]: Wow !

File: internal/db/rdb/mysql/mysql.go [hunk 2]
```diff
@@ -118,8 +116,18 @@ func (m *mySQLClient) Open(ctx context.Context) (err error) {
...
+			func() string {
+				switch net.NetworkTypeFromString(m.network) {
+				case net.UNIX, net.UNIXGRAM, net.UNIXPACKET:
+					if len(m.socketPath) != 0 {
+						return m.socketPath
+					}
+				}
+				return net.JoinHostPort(m.host, m.port)
+			}(),
+			m.name,
```
[the remark this answers is not in the document]
Comment (kpango) [reply internal/db/rdb/mysql/mysql.go:130]: no I haven't....

File: internal/db/rdb/mysql/mysql.go [hunk 3]
```diff
@@ -97,12 +98,9 @@ func (m *mySQLClient) Open(ctx context.Context) (err error) {
...
 
 	var addParam string
-
-	network := "tcp"
-
 	if m.dialerFunc != nil {
-		mysql.RegisterDialContext(network, func(ctx context.Context, addr string) (net.Conn, error) {
-			return m.dialerFunc(ctx, network, addr)
+		mysql.RegisterDialContext(m.network, func(ctx context.Context, addr string) (net.Conn, error) {
+			return m.dialerFunc(ctx, m.network, addr)
```
[the remark this answers is not in the document]
Comment (kpango) [reply internal/db/rdb/mysql/mysql.go:103]: I didn't change this line.

File: internal/db/rdb/mysql/mysql.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kpango) [reply internal/db/rdb/mysql/mysql.go]: covered

**Review by vankichi — APPROVED** [review]
Thx ! Gread Job!
LGTM

**Review by rinx — APPROVED** [review]
checked.

**Review by hlts2 — APPROVED** [review]

**Pull request merged**